keps/sig-release/2572-release-cadence/README.md

```diff
@@ -0,0 +1,1022 @@
+
+@jberkus:
+
+> @youngnick you summed it up. Having 2 years of support for a specific API snapshot is unrealistic right now for all sorts of reasons, and it wasn't even clear that it was what people actually wanted.
+
+### TODO Go faster
+
+@sebgoa:
+
+> IMHO with more releases, developers don't need to rush their features, upgrades a more bite size and it necessarily pushes for even more automation.
+>
+> So at the risk of being down voted I would argue that we have worked over the last 15 years to agree that "release early, release often" was a good idea, that creating a tight feedback loop with devs, testers and users was a very good idea.
+>
+> Theoretically we should have processes in place to be able to automatically upgrade and be able to handle even a higher cadence of releases. I could see a future were people don't upgrade that often because there are less releases and then start to fall behind one year, then two...etc.
+
+@johnbelamaric:
+
+>     I could see a future were people don't upgrade that often because there are less releases and then start to fall behind one year, then two...etc.
+>
+> Yes, this is a big fear of mine as well. We have worked hard to prevent vendor-based fragmentation (e.g., with conformance) and version-based fragmentation (with API round trip policies, etc). Bigger releases with riskier upgrades may undermine that work. We must avoid a Python2 -> 3 situation. This is also why we elected for a longer support cycle as opposed to an LTS. With the extensive ecosystem we have, fragmentation is extremely dangerous.
+>
+> I don't think going from 4->3 releases will create this problem, though I do think going to 2 or 1 release would. We need some plan around the mitigations I described earlier though, to ensure we avoid this fate.
+
+@sftim:
+
+> If there were an unsupported-but-tested Kubernetes release cut and published once a week - what would that mean?
+>
+> I'm imagining something that passes the conformance tests (little point otherwise) but comes with no guarantee. The Rust project has a model a bit like this with a daily unstable release which has nevertheless been through lots of automated testing.
+>
+> When I'm typing this I'm imagining that I could run minikube start --weekly-unstable and get a local test cluster based on the most recent release. If Kubernetes already had that built and working, would people pick different answers?
+
+@jberkus:
+
+> @sftim yeah, you've noticed that the reason, right now, we don't see a lot of community testing on alphas and betas is that we don't make them easy to consume.
+>
+> I'd say that it would need to go beyond that: we'd need images, minikube, and kubeadm releases for each weekly release.
+>
+> I don't know how that would affect our choice of major release cadence (isn't it orthagonal?) but it would be a great thing to do regardless. Also very hard.
+
+### TODO Maintenance releases
```

Comment:
I would mark this as out of scope of the KEP.

Reply:
Mark as out-of-scope in 240e8b7d6bc2e0bd12115001c197558f067451ae.
Separate discussion opened here: https://github.com/kubernetes/sig-release/discussions/1495